Crash on when shutting down the server with compile_expressions enabled

Build master/HEAD on fcff8dc3d59808b3410196143c66ae537096949d

Debug build with:
```
CC=clang CXX=clang++ cmake -DCMAKE_BUILD_TYPE=Debug ..
ninja
```

```
clang version 12.0.0
ninja version 1.10.2
```

**Describe the bug**

The server crashes during shutdown.

Started the server (Note that this is a clean path and default config):
```
./programs/clickhouse server --config-file=../programs/server/config.xml
```

Launch a stateless tests that uses JIT (on a different tab):

```
PATH=$PATH:build_head_asserts/programs/ tests/clickhouse-test 01278_min_insert_block_size_rows_for_materialized_views
```

Once the test has finished, stop the server (Ctrl+C):
```
[...]
2021.06.29 17:29:27.067302 [ 466554 ] {} <Debug> Application: Shut down storages.
2021.06.29 17:29:27.067683 [ 466554 ] {} <Debug> MemoryTracker: Peak memory usage (for user): 2.00 MiB.
2021.06.29 17:29:27.071756 [ 466554 ] {} <Debug> Application: Destroyed global context.
2021.06.29 17:29:27.081111 [ 466554 ] {} <Information> Application: shutting down
2021.06.29 17:29:27.081174 [ 466554 ] {} <Debug> Application: Uninitializing subsystem: Logging Subsystem
2021.06.29 17:29:27.081260 [ 466555 ] {} <Trace> BaseDaemon: Received signal -2
2021.06.29 17:29:27.081360 [ 466555 ] {} <Information> BaseDaemon: Stop SignalListener thread
Segmentation fault (core dumped)
```

**Does it reproduce on recent release?**

I've only tested in master/HEAD

**Enable crash reporting**

Done, no idea if it's useful:
```
2021.06.29 17:39:47.679502 [ 472860 ] {} <Information> SentryWriter: Sending crash reports is initialized with https://6f33034cfe684dd7a3ab9875e57b1c8d@o388870.ingest.sentry.io/5226277 endpoint and ./tmp/sentry temp folder
2021.06.29 17:39:47.679608 [ 472860 ] {} <Trace> Pipe: Pipe capacity is 1.00 MiB
*   Trying 34.120.195.249:443...
* Connected to o388870.ingest.sentry.io () port 443 (#0)
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: none
* SSL connection using TLSv1.3 / TLS_AES_128_GCM_SHA256
* ALPN, server accepted to use http/1.1
* Server certificate:
*  subject: CN=*.ingest.sentry.io
*  start date: Jun 26 01:07:31 2021 GMT
*  expire date: Sep 24 01:07:30 2021 GMT
*  subjectAltName: host "o388870.ingest.sentry.io" matched cert's "*.ingest.sentry.io"
*  issuer: C=US; O=Let's Encrypt; CN=R3
*  SSL certificate verify ok.
> POST /api/5226277/envelope/ HTTP/1.1
Host: o388870.ingest.sentry.io
User-Agent: sentry.native/0.3.4
Accept: */*
x-sentry-auth:Sentry sentry_key=6f33034cfe684dd7a3ab9875e57b1c8d, sentry_version=7, sentry_client=sentry.native/0.3.4
content-type:application/x-sentry-envelope
content-length:303

* upload completely sent off: 303 out of 303 bytes
* old SSL session ID is stale, removing
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Server: nginx
< Date: Tue, 29 Jun 2021 15:39:47 GMT
< Content-Type: application/json
< Content-Length: 2
< access-control-expose-headers: x-sentry-rate-limits, retry-after, x-sentry-error
< vary: Origin
< x-envoy-upstream-service-time: 0
< Strict-Transport-Security: max-age=31536000; includeSubDomains; preload
< Via: 1.1 google
< Alt-Svc: clear
< 
{}* Connection #0 to host o388870.ingest.sentry.io left intact
```

**Error message and/or stacktrace**
Coredump backtrace:
```
#0  0x000000001db46890 in std::__1::__hash_table<std::__1::__hash_value_type<unsigned long, std::__1::unique_ptr<DB::JITModuleMemoryManager, std::__1::default_delete<DB::JITModuleMemoryManager> > >, std::__1::__unordered_map_hasher<unsigned long, std::__1::__hash_value_type<unsigned long, std::__1::unique_ptr<DB::JITModuleMemoryManager, std::__1::default_delete<DB::JITModuleMemoryManager> > >, std::__1::hash<unsigned long>, std::__1::equal_to<unsigned long>, true>, std::__1::__unordered_map_equal<unsigned long, std::__1::__hash_value_type<unsigned long, std::__1::unique_ptr<DB::JITModuleMemoryManager, std::__1::default_delete<DB::JITModuleMemoryManager> > >, std::__1::equal_to<unsigned long>, std::__1::hash<unsigned long>, true>, std::__1::allocator<std::__1::__hash_value_type<unsigned long, std::__1::unique_ptr<DB::JITModuleMemoryManager, std::__1::default_delete<DB::JITModuleMemoryManager> > > > >::find<unsigned long> (this=0x26aa10c8 <DB::getJITInstance()::jit+504>, __k=@0x7f462c39c0c8: 7)
    at ../contrib/libcxx/include/__hash_table:2394
#1  0x000000001db35f8a in std::__1::unordered_map<unsigned long, std::__1::unique_ptr<DB::JITModuleMemoryManager, std::__1::default_delete<DB::JITModuleMemoryManager> >, std::__1::hash<unsigned long>, std::__1::equal_to<unsigned long>, std::__1::allocator<std::__1::pair<unsigned long const, std::__1::unique_ptr<DB::JITModuleMemoryManager, std::__1::default_delete<DB::JITModuleMemoryManager> > > > >::find (this=0x26aa10c8 <DB::getJITInstance()::jit+504>, __k=@0x7f462c39c0c8: 7)
    at ../contrib/libcxx/include/unordered_map:1352
#2  0x000000001db33ad6 in DB::CHJIT::deleteCompiledModule (this=0x26aa0ed0 <DB::getJITInstance()::jit>, module_info=...) at ../src/Interpreters/JIT/CHJIT.cpp:273
#3  0x000000001d131456 in DB::CompiledFunction::~CompiledFunction (this=0x7f462c39c0b8) at ../src/Interpreters/ExpressionJIT.cpp:58
#4  0x000000001d131419 in std::__1::allocator<DB::CompiledFunction>::destroy (this=0x7ffdfbc9faa0, __p=0x7f462c39c0b8) at ../contrib/libcxx/include/memory:891
#5  0x000000001d1313dd in std::__1::allocator_traits<std::__1::allocator<DB::CompiledFunction> >::__destroy<DB::CompiledFunction> (__a=..., __p=0x7f462c39c0b8) at ../contrib/libcxx/include/__memory/allocator_traits.h:539
#6  0x000000001d13139d in std::__1::allocator_traits<std::__1::allocator<DB::CompiledFunction> >::destroy<DB::CompiledFunction> (__a=..., __p=0x7f462c39c0b8) at ../contrib/libcxx/include/__memory/allocator_traits.h:487
#7  0x000000001d13103b in std::__1::__shared_ptr_emplace<DB::CompiledFunction, std::__1::allocator<DB::CompiledFunction> >::__on_zero_shared (this=0x7f462c39c0a0) at ../contrib/libcxx/include/memory:2611
#8  0x0000000012228951 in std::__1::__shared_count::__release_shared (this=0x7f462c39c0a0) at ../contrib/libcxx/include/memory:2475
#9  0x00000000122288f9 in std::__1::__shared_weak_count::__release_shared (this=0x7f462c39c0a0) at ../contrib/libcxx/include/memory:2517
#10 0x000000001d11ec8c in std::__1::shared_ptr<DB::CompiledFunction>::~shared_ptr (this=0x7f45e7e56af8) at ../contrib/libcxx/include/memory:3212
#11 0x000000001d12cab5 in DB::CompiledFunctionCacheEntry::~CompiledFunctionCacheEntry (this=0x7f45e7e56af8) at ../src/Interpreters/ExpressionJIT.h:16
#12 0x000000001d12ca99 in std::__1::allocator<DB::CompiledFunctionCacheEntry>::destroy (this=0x7ffdfbc9fbe0, __p=0x7f45e7e56af8) at ../contrib/libcxx/include/memory:891
#13 0x000000001d12ca5d in std::__1::allocator_traits<std::__1::allocator<DB::CompiledFunctionCacheEntry> >::__destroy<DB::CompiledFunctionCacheEntry> (__a=..., __p=0x7f45e7e56af8) at ../contrib/libcxx/include/__memory/allocator_traits.h:539
#14 0x000000001d12ca1d in std::__1::allocator_traits<std::__1::allocator<DB::CompiledFunctionCacheEntry> >::destroy<DB::CompiledFunctionCacheEntry> (__a=..., __p=0x7f45e7e56af8) at ../contrib/libcxx/include/__memory/allocator_traits.h:487
#15 0x000000001d12c73b in std::__1::__shared_ptr_emplace<DB::CompiledFunctionCacheEntry, std::__1::allocator<DB::CompiledFunctionCacheEntry> >::__on_zero_shared (this=0x7f45e7e56ae0) at ../contrib/libcxx/include/memory:2611
#16 0x0000000012228951 in std::__1::__shared_count::__release_shared (this=0x7f45e7e56ae0) at ../contrib/libcxx/include/memory:2475
#17 0x00000000122288f9 in std::__1::__shared_weak_count::__release_shared (this=0x7f45e7e56ae0) at ../contrib/libcxx/include/memory:2517
#18 0x000000001d127c0c in std::__1::shared_ptr<DB::CompiledFunctionCacheEntry>::~shared_ptr (this=0x7f4634c646e0) at ../contrib/libcxx/include/memory:3212
#19 0x000000001d12fe95 in DB::LRUCache<wide::integer<128ul, unsigned int>, DB::CompiledFunctionCacheEntry, UInt128Hash, DB::CompiledFunctionWeightFunction>::Cell::~Cell (this=0x7f4634c646e0) at ../src/Common/LRUCache.h:170
#20 0x000000001d12fe79 in std::__1::pair<wide::integer<128ul, unsigned int> const, DB::LRUCache<wide::integer<128ul, unsigned int>, DB::CompiledFunctionCacheEntry, UInt128Hash, DB::CompiledFunctionWeightFunction>::Cell>::~pair (this=0x7f4634c646d0)
    at ../contrib/libcxx/include/utility:297
#21 0x000000001d12fe2d in std::__1::destroy_at<std::__1::pair<wide::integer<128ul, unsigned int> const, DB::LRUCache<wide::integer<128ul, unsigned int>, DB::CompiledFunctionCacheEntry, UInt128Hash, DB::CompiledFunctionWeightFunction>::Cell> > (
    __loc=0x7f4634c646d0) at ../contrib/libcxx/include/__memory/base.h:118
#22 0x000000001d12fdb9 in std::__1::allocator_traits<std::__1::allocator<std::__1::__hash_node<std::__1::__hash_value_type<wide::integer<128ul, unsigned int>, DB::LRUCache<wide::integer<128ul, unsigned int>, DB::CompiledFunctionCacheEntry, UInt128Hash, DB::CompiledFunctionWeightFunction>::Cell>, void*> > >::__destroy<std::__1::pair<wide::integer<128ul, unsigned int> const, DB::LRUCache<wide::integer<128ul, unsigned int>, DB::CompiledFunctionCacheEntry, UInt128Hash, DB::CompiledFunctionWeightFunction>::Cell> > (__p=0x7f4634c646d0) at ../contrib/libcxx/include/__memory/allocator_traits.h:547
#23 0x000000001d12fd3d in std::__1::allocator_traits<std::__1::allocator<std::__1::__hash_node<std::__1::__hash_value_type<wide::integer<128ul, unsigned int>, DB::LRUCache<wide::integer<128ul, unsigned int>, DB::CompiledFunctionCacheEntry, UInt128Hash, DB::CompiledFunctionWeightFunction>::Cell>, void*> > >::destroy<std::__1::pair<wide::integer<128ul, unsigned int> const, DB::LRUCache<wide::integer<128ul, unsigned int>, DB::CompiledFunctionCacheEntry, UInt128Hash, DB::CompiledFunctionWeightFunction>::Cell> > (__a=..., __p=0x7f4634c646d0) at ../contrib/libcxx/include/__memory/allocator_traits.h:487
#24 0x000000001d133921 in std::__1::__hash_table<std::__1::__hash_value_type<wide::integer<128ul, unsigned int>, DB::LRUCache<wide::integer<128ul, unsigned int>, DB::CompiledFunctionCacheEntry, UInt128Hash, DB::CompiledFunctionWeightFunction>::Cell>, std::__1::__unordered_map_hasher<wide::integer<128ul, unsigned int>, std::__1::__hash_value_type<wide::integer<128ul, unsigned int>, DB::LRUCache<wide::integer<128ul, unsigned int>, DB::CompiledFunctionCacheEntry, UInt128Hash, DB::CompiledFunctionWeightFunction>::Cell>, UInt128Hash, std::__1::equal_to<wide::integer<128ul, unsigned int> >, true>, std::__1::__unordered_map_equal<wide::integer<128ul, unsigned int>, std::__1::__hash_value_type<wide::integer<128ul, unsigned int>, DB::LRUCache<wide::integer<128ul, unsigned int>, DB::CompiledFunctionCacheEntry, UInt128Hash, DB::CompiledFunctionWeightFunction>::Cell>, std::__1::equal_to<wide::integer<128ul, unsigned int> >, UInt128Hash, true>, std::__1::allocator<std::__1::__hash_value_type<wide::integer<128ul, unsigned int>, DB::LRUCache<wide::integer<128ul, unsigned int>, DB::CompiledFunctionCacheEntry, UInt128Hash, DB::CompiledFunctionWeightFunction>::Cell> > >::__deallocate_node (this=0x7f47001f0508, __np=0x7f4634c646c0)
    at ../contrib/libcxx/include/__hash_table:1580
#25 0x000000001d1338a9 in std::__1::__hash_table<std::__1::__hash_value_type<wide::integer<128ul, unsigned int>, DB::LRUCache<wide::integer<128ul, unsigned int>, DB::CompiledFunctionCacheEntry, UInt128Hash, DB::CompiledFunctionWeightFunction>::Cell>, std::__1::__unordered_map_hasher<wide::integer<128ul, unsigned int>, std::__1::__hash_value_type<wide::integer<128ul, unsigned int>, DB::LRUCache<wide::integer<128ul, unsigned int>, DB::CompiledFunctionCacheEntry, UInt128Hash, DB::CompiledFunctionWeightFunction>::Cell>, UInt128Hash, std::__1::equal_to<wide::integer<128ul, unsigned int> >, true>, std::__1::__unordered_map_equal<wide::integer<128ul, unsigned int>, std::__1::__hash_value_type<wide::integer<128ul, unsigned int>, DB::LRUCache<wide::integer<128ul, unsigned int>, DB::CompiledFunctionCacheEntry, UInt128Hash, DB::CompiledFunctionWeightFunction>::Cell>, std::__1::equal_to<wide::integer<128ul, unsigned int> >, UInt128Hash, true>, std::__1::allocator<std::__1::__hash_value_type<wide::integer<128ul, unsigned int>, DB::LRUCache<wide::integer<128ul, unsigned int>, DB::CompiledFunctionCacheEntry, UInt128Hash, DB::CompiledFunctionWeightFunction>::Cell> > >::~__hash_table (this=0x7f47001f0508) at ../contrib/libcxx/include/__hash_table:1519
#26 0x000000001d132bf5 in std::__1::unordered_map<wide::integer<128ul, unsigned int>, DB::LRUCache<wide::integer<128ul, unsigned int>, DB::CompiledFunctionCacheEntry, UInt128Hash, DB::CompiledFunctionWeightFunction>::Cell, UInt128Hash, std::__1::equal_to<wide::integer<128ul, unsigned int> >, std::__1::allocator<std::__1::pair<wide::integer<128ul, unsigned int> const, DB::LRUCache<wide::integer<128ul, unsigned int>, DB::CompiledFunctionCacheEntry, UInt128Hash, DB::CompiledFunctionWeightFunction>::Cell> > >::~unordered_map (this=0x7f47001f0508) at ../contrib/libcxx/include/unordered_map:1044
#27 0x000000001d132c48 in DB::LRUCache<wide::integer<128ul, unsigned int>, DB::CompiledFunctionCacheEntry, UInt128Hash, DB::CompiledFunctionWeightFunction>::~LRUCache (this=0x7f47001f0500) at ../src/Common/LRUCache.h:164
#28 0x000000001d132b15 in DB::CompiledExpressionCache::~CompiledExpressionCache (this=0x7f47001f0500) at ../src/Interpreters/ExpressionJIT.h:45
#29 0x000000001d132b39 in DB::CompiledExpressionCache::~CompiledExpressionCache (this=0x7f47001f0500) at ../src/Interpreters/ExpressionJIT.h:45
#30 0x000000001d13288c in std::__1::default_delete<DB::CompiledExpressionCache>::operator() (this=0x26aa0e58 <DB::CompiledExpressionCacheFactory::instance()::factory>, __ptr=0x7f47001f0500) at ../contrib/libcxx/include/memory:1397
#31 0x000000001d1327fc in std::__1::unique_ptr<DB::CompiledExpressionCache, std::__1::default_delete<DB::CompiledExpressionCache> >::reset (this=0x26aa0e58 <DB::CompiledExpressionCacheFactory::instance()::factory>, __p=0x0)
    at ../contrib/libcxx/include/memory:1658
#32 0x000000001d11cbd9 in std::__1::unique_ptr<DB::CompiledExpressionCache, std::__1::default_delete<DB::CompiledExpressionCache> >::~unique_ptr (this=0x26aa0e58 <DB::CompiledExpressionCacheFactory::instance()::factory>)
    at ../contrib/libcxx/include/memory:1612
#33 0x000000001d11ca55 in DB::CompiledExpressionCacheFactory::~CompiledExpressionCacheFactory (this=0x26aa0e58 <DB::CompiledExpressionCacheFactory::instance()::factory>) at ../src/Interpreters/ExpressionJIT.h:52
#34 0x00007f47015af4a7 in __run_exit_handlers () from /usr/lib/libc.so.6
#35 0x00007f47015af64e in exit () from /usr/lib/libc.so.6
#36 0x00007f4701597b2c in __libc_start_main () from /usr/lib/libc.so.6
--Type <RET> for more, q to quit, c to continue without paging--
#37 0x0000000012220bee in _start ()
```

**Additional context**

Reproduces everytime under in my system. If I disable compile_expressions in users.xml (`<compile_expressions>false</compile_expressions>`) it stops crashing and the server shuts down nicely.

Cmake config (from default parameters):

<details>

```
cmake -LA
CMake Warning:
  No source or binary directory provided.  Both will be assumed to be the
  same as the current working directory, but note that this warning will
  become a fatal error in future CMake releases.


CMake Error: The source directory "/mnt/ch/ClickHouse/build_head_asserts" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
-- Cache values
ABSL_RUN_TESTS:BOOL=OFF
ABSL_USE_GOOGLETEST_HEAD:BOOL=OFF
ADD_GDB_INDEX_FOR_GOLD:BOOL=OFF
ARCH_NATIVE:BOOL=OFF
AWK_PROGRAM:FILEPATH=/usr/bin/awk
BIN_INSTALL_DIR:PATH=/usr/local/bin
BUG_REPORT_URL:STRING=https://bugs.llvm.org/
BUILD_TESTING:BOOL=ON
Backtrace_HEADER:STRING=execinfo.h
Backtrace_INCLUDE_DIR:PATH=/usr/include
Backtrace_LIBRARY:FILEPATH=
CARES_BUILD_TESTS:BOOL=OFF
CARES_BUILD_TOOLS:BOOL=ON
CARES_INSTALL:BOOL=ON
CARES_SHARED:BOOL=OFF
CARES_STATIC:BOOL=ON
CARES_STATIC_PIC:BOOL=OFF
CASS_BUILD_EXAMPLES:BOOL=OFF
CASS_BUILD_INTEGRATION_TESTS:BOOL=OFF
CASS_BUILD_SHARED:BOOL=ON
CASS_BUILD_STATIC:BOOL=ON
CASS_BUILD_TESTS:BOOL=OFF
CASS_BUILD_UNIT_TESTS:BOOL=OFF
CASS_DEBUG_CUSTOM_ALLOC:BOOL=OFF
CASS_INSTALL_HEADER:BOOL=OFF
CASS_INSTALL_HEADER_IN_SUBDIR:BOOL=OFF
CASS_INSTALL_PKG_CONFIG:BOOL=OFF
CASS_MULTICORE_COMPILATION:BOOL=ON
CASS_USE_BOOST_ATOMIC:BOOL=OFF
CASS_USE_KERBEROS:BOOL=OFF
CASS_USE_LIBSSH2:BOOL=OFF
CASS_USE_OPENSSL:BOOL=ON
CASS_USE_STATIC_LIBS:BOOL=OFF
CASS_USE_STD_ATOMIC:BOOL=ON
CASS_USE_TIMERFD:BOOL=OFF
CASS_USE_ZLIB:BOOL=ON
CCACHE_FOUND:FILEPATH=/usr/bin/ccache
CLANG_FORMAT_EXE:FILEPATH=/usr/bin/clang-format
CMAKE_ADDR2LINE:FILEPATH=/usr/bin/addr2line
CMAKE_AR:FILEPATH=/usr/bin/ar
CMAKE_ASM_COMPILER:FILEPATH=/usr/lib/ccache/bin/clang
CMAKE_ASM_COMPILER_AR:FILEPATH=/usr/bin/llvm-ar
CMAKE_ASM_COMPILER_RANLIB:FILEPATH=/usr/bin/llvm-ranlib
CMAKE_ASM_FLAGS:STRING=
CMAKE_ASM_FLAGS_DEBUG:STRING=-g
CMAKE_ASM_FLAGS_MINSIZEREL:STRING=-Os -DNDEBUG
CMAKE_ASM_FLAGS_RELEASE:STRING=-O3 -DNDEBUG
CMAKE_ASM_FLAGS_RELWITHDEBINFO:STRING=-O2 -g -DNDEBUG
CMAKE_BUILD_TYPE:STRING=Debug
CMAKE_CONFIGURATION_TYPES:STRING=RelWithDebInfo;Debug;Release;MinSizeRel
CMAKE_CXX_COMPILER:FILEPATH=/usr/lib/ccache/bin/clang++
CMAKE_CXX_COMPILER_AR:FILEPATH=/usr/bin/llvm-ar
CMAKE_CXX_COMPILER_RANLIB:FILEPATH=/usr/bin/llvm-ranlib
CMAKE_CXX_FLAGS:STRING=
CMAKE_CXX_FLAGS_DEBUG:STRING=-g
CMAKE_CXX_FLAGS_MINSIZEREL:STRING=-Os -DNDEBUG
CMAKE_CXX_FLAGS_RELEASE:STRING=-O3 -DNDEBUG
CMAKE_CXX_FLAGS_RELWITHDEBINFO:STRING=-O2 -g -DNDEBUG
CMAKE_CXX_STANDARD:STRING=14
CMAKE_C_COMPILER:FILEPATH=/usr/lib/ccache/bin/clang
CMAKE_C_COMPILER_AR:FILEPATH=/usr/bin/llvm-ar
CMAKE_C_COMPILER_RANLIB:FILEPATH=/usr/bin/llvm-ranlib
CMAKE_C_FLAGS:STRING=
CMAKE_C_FLAGS_DEBUG:STRING=-g
CMAKE_C_FLAGS_MINSIZEREL:STRING=-Os -DNDEBUG
CMAKE_C_FLAGS_RELEASE:STRING=-O3 -DNDEBUG
CMAKE_C_FLAGS_RELWITHDEBINFO:STRING=-O2 -g -DNDEBUG
CMAKE_DEBUG_POSTFIX:STRING=d
CMAKE_DLLTOOL:FILEPATH=/usr/bin/llvm-dlltool
CMAKE_EXE_LINKER_FLAGS:STRING=
CMAKE_EXE_LINKER_FLAGS_DEBUG:STRING=
CMAKE_EXE_LINKER_FLAGS_MINSIZEREL:STRING=
CMAKE_EXE_LINKER_FLAGS_RELEASE:STRING=
CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO:STRING=
CMAKE_EXPORT_COMPILE_COMMANDS:BOOL=
CMAKE_INSTALL_BINDIR:PATH=bin
CMAKE_INSTALL_DATADIR:PATH=
CMAKE_INSTALL_DATAROOTDIR:PATH=share
CMAKE_INSTALL_DOCDIR:PATH=
CMAKE_INSTALL_INCLUDEDIR:PATH=include
CMAKE_INSTALL_INFODIR:PATH=
CMAKE_INSTALL_LIBDIR:PATH=lib
CMAKE_INSTALL_LIBEXECDIR:PATH=libexec
CMAKE_INSTALL_LOCALEDIR:PATH=
CMAKE_INSTALL_LOCALSTATEDIR:PATH=var
CMAKE_INSTALL_MANDIR:PATH=
CMAKE_INSTALL_OLDINCLUDEDIR:PATH=/usr/include
CMAKE_INSTALL_PREFIX:PATH=/usr/local
CMAKE_INSTALL_RUNSTATEDIR:PATH=
CMAKE_INSTALL_SBINDIR:PATH=sbin
CMAKE_INSTALL_SHAREDSTATEDIR:PATH=com
CMAKE_INSTALL_SYSCONFDIR:PATH=etc
CMAKE_LINKER:FILEPATH=/usr/bin/ld
CMAKE_MAKE_PROGRAM:FILEPATH=/usr/bin/ninja
CMAKE_MODULE_LINKER_FLAGS:STRING=
CMAKE_MODULE_LINKER_FLAGS_DEBUG:STRING=
CMAKE_MODULE_LINKER_FLAGS_MINSIZEREL:STRING=
CMAKE_MODULE_LINKER_FLAGS_RELEASE:STRING=
CMAKE_MODULE_LINKER_FLAGS_RELWITHDEBINFO:STRING=
CMAKE_NM:FILEPATH=/usr/bin/nm
CMAKE_OBJCOPY:FILEPATH=/usr/bin/objcopy
CMAKE_OBJDUMP:FILEPATH=/usr/bin/objdump
CMAKE_RANLIB:FILEPATH=/usr/bin/ranlib
CMAKE_READELF:FILEPATH=/usr/bin/readelf
CMAKE_SHARED_LINKER_FLAGS:STRING=
CMAKE_SHARED_LINKER_FLAGS_DEBUG:STRING=
CMAKE_SHARED_LINKER_FLAGS_MINSIZEREL:STRING=
CMAKE_SHARED_LINKER_FLAGS_RELEASE:STRING=
CMAKE_SHARED_LINKER_FLAGS_RELWITHDEBINFO:STRING=
CMAKE_SKIP_INSTALL_RPATH:BOOL=NO
CMAKE_SKIP_RPATH:BOOL=NO
CMAKE_STATIC_LINKER_FLAGS:STRING=
CMAKE_STATIC_LINKER_FLAGS_DEBUG:STRING=
CMAKE_STATIC_LINKER_FLAGS_MINSIZEREL:STRING=
CMAKE_STATIC_LINKER_FLAGS_RELEASE:STRING=
CMAKE_STATIC_LINKER_FLAGS_RELWITHDEBINFO:STRING=
CMAKE_STRIP:FILEPATH=/usr/bin/strip
CMAKE_VERBOSE_MAKEFILE:BOOL=FALSE
COMPILER_PIPE:BOOL=ON
COVERAGE_COMMAND:FILEPATH=/usr/bin/gcov
COVERAGE_EXTRA_FLAGS:STRING=-l
CPACK_BINARY_DEB:BOOL=OFF
CPACK_BINARY_FREEBSD:BOOL=OFF
CPACK_BINARY_IFW:BOOL=OFF
CPACK_BINARY_NSIS:BOOL=OFF
CPACK_BINARY_RPM:BOOL=OFF
CPACK_BINARY_STGZ:BOOL=ON
CPACK_BINARY_TBZ2:BOOL=OFF
CPACK_BINARY_TGZ:BOOL=ON
CPACK_BINARY_TXZ:BOOL=OFF
CPACK_BINARY_TZ:BOOL=ON
CPACK_SOURCE_RPM:BOOL=OFF
CPACK_SOURCE_TBZ2:BOOL=ON
CPACK_SOURCE_TGZ:BOOL=ON
CPACK_SOURCE_TXZ:BOOL=ON
CPACK_SOURCE_TZ:BOOL=ON
CPACK_SOURCE_ZIP:BOOL=OFF
CTEST_SUBMIT_RETRY_COUNT:STRING=3
CTEST_SUBMIT_RETRY_DELAY:STRING=5
CURL_DIR:PATH=CURL_DIR-NOTFOUND
CURL_FOUND:BOOL=ON
CURL_INCLUDE_DIR:PATH=/mnt/ch/ClickHouse/contrib/curl/include
CURL_INCLUDE_DIRS:PATH=/mnt/ch/ClickHouse/contrib/curl/include
CURL_LIBRARIES:STRING=curl
CURL_LIBRARY:STRING=curl
CURL_ROOT_DIR:PATH=/mnt/ch/ClickHouse/contrib/curl
CURL_VERSION_STRING:STRING=7.67.0
DART_TESTING_TIMEOUT:STRING=1500
ENABLED_LOCAL_INFILE:STRING=AUTO
ENABLE_AMQPCPP:BOOL=ON
ENABLE_AVRO:BOOL=ON
ENABLE_BASE64:BOOL=ON
ENABLE_BROTLI:BOOL=ON
ENABLE_CAPNP:BOOL=ON
ENABLE_CASSANDRA:BOOL=ON
ENABLE_CCACHE:BOOL=ON
ENABLE_CHECK_HEAVY_BUILDS:BOOL=OFF
ENABLE_CLANG_TIDY:BOOL=OFF
ENABLE_CLICKHOUSE_ALL:BOOL=ON
ENABLE_CLICKHOUSE_BENCHMARK:BOOL=ON
ENABLE_CLICKHOUSE_CLIENT:BOOL=ON
ENABLE_CLICKHOUSE_COMPRESSOR:BOOL=ON
ENABLE_CLICKHOUSE_COPIER:BOOL=ON
ENABLE_CLICKHOUSE_EXTRACT_FROM_CONFIG:BOOL=ON
ENABLE_CLICKHOUSE_FORMAT:BOOL=ON
ENABLE_CLICKHOUSE_GIT_IMPORT:BOOL=ON
ENABLE_CLICKHOUSE_INSTALL:BOOL=ON
ENABLE_CLICKHOUSE_KEEPER:BOOL=ON
ENABLE_CLICKHOUSE_KEEPER_CONVERTER:BOOL=ON
ENABLE_CLICKHOUSE_LIBRARY_BRIDGE:BOOL=ON
ENABLE_CLICKHOUSE_LOCAL:BOOL=ON
ENABLE_CLICKHOUSE_OBFUSCATOR:BOOL=ON
ENABLE_CLICKHOUSE_ODBC_BRIDGE:BOOL=ON
ENABLE_CLICKHOUSE_SERVER:BOOL=ON
ENABLE_CPUID:BOOL=ON
ENABLE_CURL:BOOL=ON
ENABLE_CYRUS_SASL:BOOL=ON
ENABLE_DATASKETCHES:BOOL=ON
ENABLE_EMBEDDED_COMPILER:BOOL=ON
ENABLE_EXAMPLES:BOOL=OFF
ENABLE_EXPERIMENTAL_NEW_PASS_MANAGER:BOOL=FALSE
ENABLE_FASTOPS:BOOL=ON
ENABLE_FUZZING:BOOL=OFF
ENABLE_GRPC:BOOL=ON
ENABLE_GSASL_LIBRARY:BOOL=ON
ENABLE_H3:BOOL=ON
ENABLE_HDFS:BOOL=ON
ENABLE_HYPERSCAN:BOOL=ON
ENABLE_ICU:BOOL=ON
ENABLE_JEMALLOC:BOOL=ON
ENABLE_KRB5:BOOL=ON
ENABLE_LDAP:BOOL=ON
ENABLE_LIBPQXX:BOOL=ON
ENABLE_LIBRARIES:BOOL=ON
ENABLE_MSGPACK:BOOL=ON
ENABLE_MULTITARGET_CODE:BOOL=ON
ENABLE_MYSQL:BOOL=ON
ENABLE_NURAFT:BOOL=ON
ENABLE_ODBC:BOOL=ON
ENABLE_ORC:BOOL=ON
ENABLE_PARQUET:BOOL=ON
ENABLE_PROTOBUF:BOOL=ON
ENABLE_RAPIDJSON:BOOL=ON
ENABLE_RDKAFKA:BOOL=ON
ENABLE_REPLXX:BOOL=ON
ENABLE_ROCKSDB:BOOL=ON
ENABLE_S3:BOOL=ON
ENABLE_SSE:BOOL=ON
ENABLE_SSL:BOOL=ON
ENABLE_STATS:BOOL=ON
ENABLE_TESTS:BOOL=ON
FAIL_ON_UNSUPPORTED_OPTIONS_COMBINATION:BOOL=ON
FASTFLOAT_SANITIZE:BOOL=OFF
FFI_INCLUDE_DIR:PATH=
FFI_LIBRARY_DIR:PATH=
FLATBUFFERS_BUILD_CPP17:BOOL=OFF
FLATBUFFERS_BUILD_FLATC:BOOL=ON
FLATBUFFERS_BUILD_FLATHASH:BOOL=ON
FLATBUFFERS_BUILD_FLATLIB:BOOL=ON
FLATBUFFERS_BUILD_GRPCTEST:BOOL=OFF
FLATBUFFERS_BUILD_LEGACY:BOOL=OFF
FLATBUFFERS_BUILD_SHAREDLIB:BOOL=OFF
FLATBUFFERS_BUILD_TESTS:BOOL=OFF
FLATBUFFERS_CODE_COVERAGE:BOOL=OFF
FLATBUFFERS_CODE_SANITIZE:BOOL=OFF
FLATBUFFERS_ENABLE_PCH:BOOL=OFF
FLATBUFFERS_INSTALL:BOOL=ON
FLATBUFFERS_LIBCXX_WITH_CLANG:BOOL=ON
FLATBUFFERS_PACKAGE_DEBIAN:BOOL=OFF
FLATBUFFERS_PACKAGE_REDHAT:BOOL=OFF
FLATBUFFERS_STATIC_FLATC:BOOL=OFF
GCEM_BUILD_TESTS:BOOL=OFF
GCEM_CMAKECONFIG_INSTALL_DIR:STRING=lib/cmake/gcem
GIT:FILEPATH=/usr/bin/git
GITCOMMAND:FILEPATH=/usr/bin/git
GIT_EXECUTABLE:FILEPATH=/usr/bin/git
GLIBC_COMPATIBILITY:BOOL=ON
GOLD_EXECUTABLE:FILEPATH=/usr/bin/ld.gold
GOLD_PATH:FILEPATH=/usr/bin/ld.gold
GO_EXECUTABLE:FILEPATH=/usr/bin/go
GSSAPI_FLAVOR:STRING=MIT
GSSAPI_FOUND:STRING=TRUE
GSSAPI_INCS:STRING=
GSSAPI_LIBS:STRING=-lgssapi_krb5 -lkrb5 -lk5crypto -lcom_err
GTest_DIR:PATH=/usr/lib64/cmake/GTest
Gflags_DIR:PATH=Gflags_DIR-NOTFOUND
HAVE_LIBNSL:BOOL=OFF
ICONV_INCLUDE_DIR:PATH=/usr/include
ICONV_LIBRARIES:FILEPATH=ICONV_LIBRARIES-NOTFOUND
INC_INSTALL_DIR:PATH=/usr/local/include
INSTALL_LAYOUT:STRING=DEFAULT
INSTALL_UTILS:BOOL=OFF
JEMALLOC_CONFIG_MALLOC_CONF_OVERRIDE:STRING=
KRB5_CONFIG:FILEPATH=/usr/bin/krb5-config
LIBRT:FILEPATH=/usr/lib/librt.a
LIB_INSTALL_DIR:PATH=/usr/local/lib
LINKER_NAME:BOOL=OFF
LLD_PATH:FILEPATH=/usr/bin/ld.lld
LLVM_ABI_BREAKING_CHECKS:STRING=WITH_ASSERTS
LLVM_APPEND_VC_REV:BOOL=ON
LLVM_AR_PATH:FILEPATH=/usr/bin/llvm-ar
LLVM_BINUTILS_INCDIR:PATH=
LLVM_BUILD_32_BITS:BOOL=OFF
LLVM_BUILD_BENCHMARKS:BOOL=OFF
LLVM_BUILD_DOCS:BOOL=OFF
LLVM_BUILD_EXAMPLES:BOOL=OFF
LLVM_BUILD_EXTERNAL_COMPILER_RT:BOOL=OFF
LLVM_BUILD_INSTRUMENTED:STRING=OFF
LLVM_BUILD_INSTRUMENTED_COVERAGE:BOOL=OFF
LLVM_BUILD_LLVM_C_DYLIB:BOOL=OFF
LLVM_BUILD_LLVM_DYLIB:BOOL=OFF
LLVM_BUILD_RUNTIME:BOOL=OFF
LLVM_BUILD_RUNTIMES:BOOL=OFF
LLVM_BUILD_TESTS:BOOL=OFF
LLVM_BUILD_TOOLS:BOOL=OFF
LLVM_BUILD_UTILS:BOOL=OFF
LLVM_CCACHE_BUILD:BOOL=OFF
LLVM_CODESIGNING_IDENTITY:STRING=
LLVM_DEFAULT_TARGET_TRIPLE:STRING=x86_64-unknown-linux-gnu
LLVM_DEPENDENCY_DEBUGGING:BOOL=OFF
LLVM_DYLIB_COMPONENTS:STRING=all
LLVM_ENABLE_ASSERTIONS:BOOL=ON
LLVM_ENABLE_BACKTRACES:BOOL=OFF
LLVM_ENABLE_BINDINGS:BOOL=OFF
LLVM_ENABLE_CRASH_DUMPS:BOOL=OFF
LLVM_ENABLE_CRASH_OVERRIDES:BOOL=OFF
LLVM_ENABLE_DAGISEL_COV:BOOL=OFF
LLVM_ENABLE_DOXYGEN:BOOL=OFF
LLVM_ENABLE_DUMP:BOOL=OFF
LLVM_ENABLE_EXPENSIVE_CHECKS:BOOL=OFF
LLVM_ENABLE_FFI:BOOL=OFF
LLVM_ENABLE_GISEL_COV:BOOL=OFF
LLVM_ENABLE_IDE:BOOL=ON
LLVM_ENABLE_IR_PGO:BOOL=OFF
LLVM_ENABLE_LIBCXX:BOOL=OFF
LLVM_ENABLE_LIBEDIT:BOOL=OFF
LLVM_ENABLE_LIBPFM:BOOL=OFF
LLVM_ENABLE_LIBXML2:STRING=OFF
LLVM_ENABLE_LLD:BOOL=OFF
LLVM_ENABLE_LOCAL_SUBMODULE_VISIBILITY:BOOL=ON
LLVM_ENABLE_LTO:STRING=OFF
LLVM_ENABLE_MODULES:BOOL=OFF
LLVM_ENABLE_MODULE_DEBUGGING:BOOL=OFF
LLVM_ENABLE_OCAMLDOC:BOOL=OFF
LLVM_ENABLE_PEDANTIC:BOOL=ON
LLVM_ENABLE_PER_TARGET_RUNTIME_DIR:BOOL=OFF
LLVM_ENABLE_PLUGINS:BOOL=OFF
LLVM_ENABLE_PROJECTS:STRING=
LLVM_ENABLE_PROJECTS_USED:BOOL=OFF
LLVM_ENABLE_SPHINX:BOOL=OFF
LLVM_ENABLE_STRICT_FIXED_SIZE_VECTORS:BOOL=OFF
LLVM_ENABLE_TERMINFO:BOOL=OFF
LLVM_ENABLE_THREADS:BOOL=ON
LLVM_ENABLE_UNWIND_TABLES:BOOL=ON
LLVM_ENABLE_WARNINGS:BOOL=ON
LLVM_ENABLE_WERROR:BOOL=OFF
LLVM_ENABLE_Z3_SOLVER:BOOL=OFF
LLVM_ENABLE_ZLIB:STRING=OFF
LLVM_EXPERIMENTAL_TARGETS_TO_BUILD:STRING=
LLVM_EXPORT_SYMBOLS_FOR_PLUGINS:BOOL=OFF
LLVM_EXTERNALIZE_DEBUGINFO:BOOL=OFF
LLVM_FORCE_ENABLE_STATS:BOOL=OFF
LLVM_FORCE_USE_OLD_TOOLCHAIN:BOOL=OFF
LLVM_HOST_TRIPLE:STRING=x86_64-unknown-linux-gnu
LLVM_INCLUDE_BENCHMARKS:BOOL=OFF
LLVM_INCLUDE_DOCS:BOOL=OFF
LLVM_INCLUDE_EXAMPLES:BOOL=OFF
LLVM_INCLUDE_GO_TESTS:BOOL=OFF
LLVM_INCLUDE_RUNTIMES:BOOL=OFF
LLVM_INCLUDE_TESTS:BOOL=OFF
LLVM_INCLUDE_TOOLS:BOOL=OFF
LLVM_INCLUDE_UTILS:BOOL=OFF
LLVM_INSTALL_BINUTILS_SYMLINKS:BOOL=OFF
LLVM_INSTALL_CCTOOLS_SYMLINKS:BOOL=OFF
LLVM_INSTALL_DOXYGEN_HTML_DIR:STRING=share/doc/llvm/doxygen-html
LLVM_INSTALL_MODULEMAPS:BOOL=OFF
LLVM_INSTALL_OCAMLDOC_HTML_DIR:STRING=share/doc/llvm/ocaml-html
LLVM_INSTALL_TOOLCHAIN_ONLY:BOOL=OFF
LLVM_INSTALL_UTILS:BOOL=OFF
LLVM_INTEGRATED_CRT_ALLOC:PATH=
LLVM_LIBDIR_SUFFIX:STRING=
LLVM_LIB_FUZZING_ENGINE:PATH=
LLVM_LINK_LLVM_DYLIB:BOOL=OFF
LLVM_LIT_ARGS:STRING=-sv
LLVM_LOCAL_RPATH:FILEPATH=
LLVM_OPTIMIZED_TABLEGEN:BOOL=OFF
LLVM_OPTIMIZE_SANITIZED_BUILDS:BOOL=ON
LLVM_PARALLEL_COMPILE_JOBS:STRING=
LLVM_PARALLEL_LINK_JOBS:STRING=
LLVM_PROFDATA_FILE:FILEPATH=
LLVM_RANLIB_PATH:FILEPATH=/usr/bin/llvm-ranlib
LLVM_SOURCE_PREFIX:STRING=
LLVM_SRPM_USER_BINARY_SPECFILE:FILEPATH=/mnt/ch/ClickHouse/contrib/llvm/llvm/llvm.spec.in
LLVM_STATIC_LINK_CXX_STDLIB:BOOL=OFF
LLVM_TABLEGEN:STRING=llvm-tblgen
LLVM_TARGETS_TO_BUILD:STRING=X86;AArch64
LLVM_TARGET_ARCH:STRING=host
LLVM_TARGET_TRIPLE_ENV:STRING=
LLVM_TEMPORARILY_ALLOW_OLD_TOOLCHAIN:BOOL=OFF
LLVM_TOOLS_INSTALL_DIR:STRING=bin
LLVM_UBSAN_FLAGS:STRING=-fsanitize=undefined -fno-sanitize=vptr,function -fno-sanitize-recover=all
LLVM_USE_FOLDERS:BOOL=ON
LLVM_USE_INTEL_JITEVENTS:BOOL=OFF
LLVM_USE_NEWPM:BOOL=OFF
LLVM_USE_OPROFILE:BOOL=OFF
LLVM_USE_PERF:BOOL=OFF
LLVM_USE_RELATIVE_PATHS_IN_DEBUG_INFO:BOOL=OFF
LLVM_USE_RELATIVE_PATHS_IN_FILES:BOOL=OFF
LLVM_USE_SANITIZER:STRING=
LLVM_USE_SPLIT_DWARF:BOOL=OFF
LLVM_UTILS_INSTALL_DIR:STRING=bin
LLVM_VERSION_PRINTER_SHOW_HOST_TARGET_INFO:BOOL=ON
LLVM_VP_COUNTERS_PER_SITE:STRING=1.5
LLVM_Z3_INSTALL_DIR:STRING=
MAKECOMMAND:STRING=/usr/bin/cmake --build . --config "${CTEST_CONFIGURATION_TYPE}"
MAKE_STATIC_LIBRARIES:BOOL=ON
MEMORYCHECK_COMMAND:FILEPATH=/usr/bin/valgrind
MEMORYCHECK_SUPPRESSIONS_FILE:FILEPATH=
NINJA_PATH:FILEPATH=/usr/bin/ninja
NINJA_VERSION:STRING=1.10.2
OBJCOPY_PATH:FILEPATH=/usr/bin/llvm-objcopy
OCAMLFIND:FILEPATH=OCAMLFIND-NOTFOUND
PARALLEL_COMPILE_JOBS:BOOL=OFF
PARALLEL_LINK_JOBS:BOOL=OFF
PKGCONFIG_INSTALL_DIR:PATH=/usr/local/lib/pkgconfig
PKG_CONFIG_EXECUTABLE:FILEPATH=/usr/bin/pkg-config
PY_PYGMENTS_FOUND:BOOL=OFF
PY_PYGMENTS_LEXERS_C_CPP_FOUND:BOOL=OFF
PY_YAML_FOUND:BOOL=OFF
SANITIZE:BOOL=OFF
SENTRY_BACKEND:STRING=none
SENTRY_BUILD_EXAMPLES:BOOL=OFF
SENTRY_BUILD_FORCE32:BOOL=OFF
SENTRY_BUILD_TESTS:BOOL=OFF
SENTRY_ENABLE_INSTALL:BOOL=OFF
SENTRY_EXPORT_SYMBOLS:BOOL=OFF
SENTRY_LINK_PTHREAD:BOOL=OFF
SENTRY_PIC:BOOL=OFF
SENTRY_TRANSPORT:STRING=curl
SITE:STRING=Mordor
SNAPPY_REQUIRE_AVX:BOOL=OFF
SNAPPY_REQUIRE_AVX2:BOOL=OFF
STRIP_DEBUG_SYMBOLS_FUNCTIONS:BOOL=OFF
TENSORFLOW_AOT_PATH:PATH=
TENSORFLOW_C_LIB_PATH:PATH=
TEST_HDFS_PREFIX:STRING=./
TUKLIB_FAST_UNALIGNED_ACCESS:BOOL=ON
TUKLIB_USE_UNSAFE_TYPE_PUNNING:BOOL=OFF
UNBUNDLED:BOOL=OFF
USEPCRE:BOOL=OFF
USE_AWS_MEMORY_MANAGEMENT:BOOL=OFF
USE_INCLUDE_WHAT_YOU_USE:BOOL=OFF
USE_INTERNAL_AVRO_LIBRARY:BOOL=ON
USE_INTERNAL_AWS_S3_LIBRARY:BOOL=ON
USE_INTERNAL_BOOST_LIBRARY:BOOL=ON
USE_INTERNAL_BROTLI_LIBRARY:BOOL=ON
USE_INTERNAL_CAPNP_LIBRARY:BOOL=ON
USE_INTERNAL_CCTZ_LIBRARY:BOOL=ON
USE_INTERNAL_CURL:BOOL=ON
USE_INTERNAL_DATASKETCHES_LIBRARY:BOOL=ON
USE_INTERNAL_DOUBLE_CONVERSION_LIBRARY:BOOL=ON
USE_INTERNAL_FARMHASH_LIBRARY:BOOL=ON
USE_INTERNAL_GRPC_LIBRARY:BOOL=ON
USE_INTERNAL_GTEST_LIBRARY:BOOL=ON
USE_INTERNAL_H3_LIBRARY:BOOL=ON
USE_INTERNAL_HDFS3_LIBRARY:BOOL=ON
USE_INTERNAL_HYPERSCAN_LIBRARY:BOOL=ON
USE_INTERNAL_ICU_LIBRARY:BOOL=ON
USE_INTERNAL_LDAP_LIBRARY:BOOL=ON
USE_INTERNAL_LIBCXX_LIBRARY:BOOL=ON
USE_INTERNAL_LIBGSASL_LIBRARY:BOOL=ON
USE_INTERNAL_LIBXML2_LIBRARY:BOOL=ON
USE_INTERNAL_LZ4_LIBRARY:BOOL=ON
USE_INTERNAL_MSGPACK_LIBRARY:BOOL=ON
USE_INTERNAL_MYSQL_LIBRARY:BOOL=ON
USE_INTERNAL_ODBC_LIBRARY:BOOL=ON
USE_INTERNAL_ORC_LIBRARY:BOOL=ON
USE_INTERNAL_PARQUET_LIBRARY:BOOL=ON
USE_INTERNAL_POCO_LIBRARY:BOOL=ON
USE_INTERNAL_PROTOBUF_LIBRARY:BOOL=ON
USE_INTERNAL_RAPIDJSON_LIBRARY:BOOL=ON
USE_INTERNAL_RDKAFKA_LIBRARY:BOOL=ON
USE_INTERNAL_RE2_LIBRARY:BOOL=ON
USE_INTERNAL_REPLXX_LIBRARY:BOOL=ON
USE_INTERNAL_ROCKSDB_LIBRARY:BOOL=ON
USE_INTERNAL_SNAPPY_LIBRARY:BOOL=ON
USE_INTERNAL_SPARSEHASH_LIBRARY:BOOL=ON
USE_INTERNAL_SSL_LIBRARY:BOOL=ON
USE_INTERNAL_XZ_LIBRARY:BOOL=ON
USE_INTERNAL_ZLIB_LIBRARY:BOOL=ON
USE_INTERNAL_ZSTD_LIBRARY:BOOL=ON
USE_LIBCXX:BOOL=ON
USE_SENTRY:BOOL=ON
USE_SIMDJSON:BOOL=ON
USE_SNAPPY:BOOL=ON
USE_STATIC_LIBRARIES:BOOL=ON
USE_UNWIND:BOOL=ON
USE_YAML_CPP:BOOL=ON
VERSION_EXTRA:STRING=
VERSION_TWEAK:STRING=
WERROR:BOOL=ON
WEVERYTHING:BOOL=ON
WITH_ASAN_OPTION:BOOL=OFF
WITH_AVX2:BOOL=ON
WITH_CODE_COVERAGE:BOOL=OFF
WITH_COVERAGE:BOOL=OFF
WITH_DYNCOL:BOOL=ON
WITH_FALLOCATE:BOOL=ON
WITH_FOLLY_DISTRIBUTED_MUTEX:BOOL=ON
WITH_FUZZERS:BOOL=OFF
WITH_GZFILEOP:BOOL=ON
WITH_INFLATE_ALLOW_INVALID_DIST:BOOL=OFF
WITH_INFLATE_STRICT:BOOL=OFF
WITH_JEMALLOC:BOOL=OFF
WITH_LZ4:BOOL=ON
WITH_MAINTAINER_WARNINGS:BOOL=OFF
WITH_MYSQLCOMPAT:BOOL=OFF
WITH_NEW_STRATEGIES:BOOL=ON
WITH_OPTIM:BOOL=ON
WITH_PCLMULQDQ:BOOL=ON
WITH_SANITIZER:BOOL=OFF
WITH_SNAPPY:BOOL=ON
WITH_SSE2:BOOL=ON
WITH_SSE4:BOOL=ON
WITH_SSL:BOOL=ON
WITH_SSSE3:BOOL=ON
WITH_TSAN_OPTION:BOOL=OFF
WITH_UNALIGNED:BOOL=ON
WITH_UNIT_TESTS:BOOL=ON
WITH_ZLIB:BOOL=ON
WITH_ZSTD:BOOL=ON
ZLIB_DUAL_LINK:BOOL=OFF
gRPC_ABSL_PROVIDER:STRING=clickhouse
gRPC_BACKWARDS_COMPATIBILITY_MODE:BOOL=OFF
gRPC_BUILD_CODEGEN:BOOL=ON
gRPC_BUILD_GRPC_CPP_PLUGIN:BOOL=ON
gRPC_BUILD_GRPC_CSHARP_PLUGIN:BOOL=ON
gRPC_BUILD_GRPC_NODE_PLUGIN:BOOL=ON
gRPC_BUILD_GRPC_OBJECTIVE_C_PLUGIN:BOOL=ON
gRPC_BUILD_GRPC_PHP_PLUGIN:BOOL=ON
gRPC_BUILD_GRPC_PYTHON_PLUGIN:BOOL=ON
gRPC_BUILD_GRPC_RUBY_PLUGIN:BOOL=ON
gRPC_BUILD_TESTS:BOOL=OFF
gRPC_CARES_PROVIDER:STRING=module
gRPC_INSTALL:BOOL=OFF
gRPC_INSTALL_BINDIR:STRING=bin
gRPC_INSTALL_CMAKEDIR:STRING=lib/cmake/grpc
gRPC_INSTALL_INCLUDEDIR:STRING=include
gRPC_INSTALL_LIBDIR:STRING=lib
gRPC_INSTALL_SHAREDIR:STRING=share/grpc
gRPC_PROTOBUF_PACKAGE_TYPE:STRING=
gRPC_PROTOBUF_PROVIDER:STRING=clickhouse
gRPC_RE2_PROVIDER:STRING=clickhouse
gRPC_SSL_PROVIDER:STRING=clickhouse
gRPC_USE_PROTO_LITE:BOOL=OFF
gRPC_ZLIB_PROVIDER:STRING=clickhouse
gtest_build_samples:BOOL=OFF
gtest_build_tests:BOOL=OFF
gtest_disable_pthreads:BOOL=OFF
gtest_force_shared_crt:BOOL=OFF
gtest_hide_internal_symbols:BOOL=OFF
liblzma_INSTALL_CMAKEDIR:STRING=lib/cmake/liblzma
pkg-config:FILEPATH=pkg-config-NOTFOUND
pkgcfg_lib_PC_CURL_curl:FILEPATH=/usr/lib/libcurl.so
pkgcfg_lib__LIBUV_dl:FILEPATH=/usr/lib/libdl.a
pkgcfg_lib__LIBUV_uv:FILEPATH=/usr/lib/libuv.so
pkgcfg_lib__OPENSSL_crypto:FILEPATH=/usr/lib/libcrypto.so
pkgcfg_lib__OPENSSL_ssl:FILEPATH=/usr/lib/libssl.so
protobuf_BUILD_CONFORMANCE:BOOL=OFF
protobuf_BUILD_EXAMPLES:BOOL=OFF
protobuf_BUILD_PROTOC_BINARIES:BOOL=ON
protobuf_DEBUG_POSTFIX:STRING=d
protobuf_MODULE_COMPATIBLE:BOOL=OFF
protobuf_MSVC_STATIC_RUNTIME:BOOL=ON
protobuf_VERBOSE:BOOL=OFF
```

</details>

--- a/src/Interpreters/Context.cpp
+++ b/src/Interpreters/Context.cpp
@@ -457,6 +457,11 @@ struct ContextSharedPart
         {
             auto lock = std::lock_guard(mutex);
 
+#if USE_EMBEDDED_COMPILER
+            if (auto * cache = CompiledExpressionCacheFactory::instance().tryGetCache())
+                cache->reset();
+#endif
+
             /// Preemptive destruction is important, because these objects may have a refcount to ContextShared (cyclic reference).
             /// TODO: Get rid of this.
 

--- a/src/Interpreters/Context.cpp
+++ b/src/Interpreters/Context.cpp
@@ -457,6 +457,9 @@ struct ContextSharedPart
         {
             auto lock = std::lock_guard(mutex);
 
+        /** Compiled expressions stored in cache need to be destroyed before destruction of static objects.
+          * Because CHJIT instance can be static object.
+          */
 #if USE_EMBEDDED_COMPILER
             if (auto * cache = CompiledExpressionCacheFactory::instance().tryGetCache())
                 cache->reset();

